File: backend/src/main/java/com/hardware/erp/labour/service/AttendanceService.java
package com.hardware.erp.labour.service;

import com.hardware.erp.labour.dto.AttendanceMarkRequest;
import com.hardware.erp.labour.dto.WorkerAttendanceResponse;

import java.time.LocalDate;
import java.util.List;

public interface AttendanceService {

    /** Marks (creates or corrects) attendance for one or more workers on one day, in a single transaction. */
    List<WorkerAttendanceResponse> mark(AttendanceMarkRequest request);

    List<WorkerAttendanceResponse> forDate(LocalDate date);

    List<WorkerAttendanceResponse> historyForWorker(Long workerId, LocalDate fromDate, LocalDate toDate);
}

File: frontend/src/modules/labour/services/attendanceService.ts
import { apiGet, apiPost } from '@/services/apiClient';
import type { AttendanceMarkRequest, WorkerAttendanceResponse } from '../types';

/** Backend: labour/controller/AttendanceController.java */
export const attendanceService = {
  mark: (body: AttendanceMarkRequest) =>
    apiPost<WorkerAttendanceResponse[]>('/v1/attendance', body),

  forDate: (date: string) =>
    apiGet<WorkerAttendanceResponse[]>('/v1/attendance', { params: { date } }),

  historyForWorker: (workerId: number, fromDate?: string, toDate?: string) =>
    apiGet<WorkerAttendanceResponse[]>(`/v1/attendance/worker/${workerId}`, { params: { fromDate, toDate } }),
};
